exec/java-exec/src/main/java/org/apache/drill/exec/store/ActionOnFile.java

```diff
@@ -0,0 +1,68 @@
+package org.apache.drill.exec.store;
+
+import org.apache.drill.common.config.CommonConstants;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * The action on the {@link CommonConstants#STORAGE_PLUGINS_OVERRIDE_CONF} file being performed after it's use
+ */
+public enum ActionOnFile {
+
+  NONE {
+    @Override
+    void action(URL url) {
+      // nothing to do
+    }
+  },
+  RENAME {
+    @Override
+    void action(URL url) {
+      File pluginsOverrideFile = new File(url.getPath());
+      String oldName = CommonConstants.STORAGE_PLUGINS_OVERRIDE_CONF;
+      String currentDateTime = new SimpleDateFormat("yyyyMMddHHmmss").format(new Date());
+      String newFileName = new StringBuilder(oldName)
+          .insert(oldName.lastIndexOf("."), "-" + currentDateTime)
+          .toString();
+      Path pluginsOverrideFilePath  = pluginsOverrideFile.toPath();
+      try {
+        Files.move(pluginsOverrideFilePath, pluginsOverrideFilePath.resolveSibling(newFileName));
+      } catch (IOException e) {
+        logger.error("%s file is not renamed after it's use", CommonConstants.STORAGE_PLUGINS_OVERRIDE_CONF, e);
```

Comment:
Re-phrase, for example: `There was an error during file %s rename.` Here and below.

Reply:
Done